Better explain some client.rb settings

tas50 opened the pull request:
In adding the chef_client_config resource I realized that some of these are not well spelled out and don't reflect what we want users adding.

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Better explain some client.rb settings

In adding the chef_client_config resource I realized that some of these are not well spelled out and don't reflect what we want users adding.

Signed-off-by: Tim Smith <tsmith@chef.io>

phiggins commented on content/config_rb_client.md:
```diff
@@ -394,8 +393,11 @@ This configuration file has the following settings:
+    Chef Infra Client in daemonized mode. It is recommended not to run
+    Chef Infra Client in daemonized mode and to instead rely on scheduled
+    Chef Infra Client execution from system schedulers like systemd timers,
+    cron jobs, or Windows Scheduled Tasks. Default value: `1800`.
```
```suggestion
:   The frequency (in seconds) at which Chef Infra Client runs when running
    in daemonized mode. It is recommended to not run
    in daemonized mode and to instead rely on scheduled
    execution from system schedulers like systemd timers,
    cron jobs, or Windows Scheduled Tasks. Default value: `1800`.
```

This reads kind of awkwardly repeating "Chef Infra Client" so often.

phiggins commented on content/config_rb_client.md:
```diff
@@ -444,13 +446,16 @@ This configuration file has the following settings:
-    integration testing to speed up test cycles.
+:   Run a minimal set of Ohai plugins providing data necessary for the
+    execution of Chef Infra Client's built-in resources. Setting this
+    to true will skip many large and time consuming data sets such as
```
```suggestion
    to true will skip many large and time consuming plugins such as
```

tas50 committed:
Update content/config_rb_client.md

Signed-off-by: Tim Smith <tsmith@chef.io>

Co-authored-by: pete higgins <pete@peterhiggins.org>

tas50 committed:
Update content/config_rb_client.md

Signed-off-by: Tim Smith <tsmith@chef.io>

Co-authored-by: pete higgins <pete@peterhiggins.org>

tas50 committed:
Update example text to not set incorrect logging

You shouldn't ever set info logging. Automatic is what you want and that's the default. Just leave it as is.

Signed-off-by: Tim Smith <tsmith@chef.io>

IanMadd commented on content/config_rb_client.md:
```diff
@@ -394,8 +393,11 @@ This configuration file has the following settings:
+    in daemonized mode. It is recommended to not run
+    in daemonized mode and to instead rely on scheduled
+    execution from system schedulers like systemd timers,
+    cron jobs, or Windows Scheduled Tasks. Default value: `1800`.
```
```suggestion
    in daemonized mode. We do not recommend running
    in daemonized mode. Instead you should rely on scheduled
    execution from system schedulers like systemd timers,
    cron jobs, or Windows Scheduled Tasks. Default value: `1800`.
```

IanMadd commented on content/config_rb_client.md:
```diff
@@ -444,13 +446,16 @@ This configuration file has the following settings:
+:   Run a minimal set of Ohai plugins providing data necessary for the
+    execution of Chef Infra Client's built-in resources. Setting this
+    to true will skip many large and time consuming plugins such as
+    `cloud` or `packages`. Setting this this to true may break cookbooks
```
```suggestion
    `cloud` or `packages`. Setting this to true may break cookbooks
```

IanMadd commented on content/config_rb_client.md:
```diff
@@ -444,13 +446,16 @@ This configuration file has the following settings:
 
-:   The run-list associated with a policy file.
+:   A specific named runlist defined in the node's applied Policyfile,
+    which the should be used when running Chef Infra Client.
```
```suggestion
:   A specific named runlist defined in the node's applied Policyfile
    which should be used when running Chef Infra Client.
```

IanMadd commented on content/config_rb_client.md:
```diff
@@ -464,11 +469,11 @@ This configuration file has the following settings:
-    authenticating to a Chef Infra Server. The default value is the Chef
-    Infra Client FQDN, as detected by Ohai. In general, Chef recommends
-    that you leave this setting blank and let Ohai assign the FQDN of
+    authenticating to a Chef Infra Server. By default Chef Infra Client
```
```suggestion
    authenticating to a Chef Infra Server. By default, Chef Infra Client
```

tas50 committed:
Update content/config_rb_client.md

Signed-off-by: Tim Smith <tsmith@chef.io>

Co-authored-by: Ian Maddaus <IanMadd@users.noreply.github.com>

tas50 committed:
Update content/config_rb_client.md

Signed-off-by: Tim Smith <tsmith@chef.io>

Co-authored-by: Ian Maddaus <IanMadd@users.noreply.github.com>

tas50 committed:
Update content/config_rb_client.md

Signed-off-by: Tim Smith <tsmith@chef.io>

Co-authored-by: Ian Maddaus <IanMadd@users.noreply.github.com>

tas50 committed:
Update content/config_rb_client.md

Signed-off-by: Tim Smith <tsmith@chef.io>

Co-authored-by: Ian Maddaus <IanMadd@users.noreply.github.com>

lamont-granquist commented on content/config_rb_client.md:
```diff
@@ -430,9 +432,9 @@ This configuration file has the following settings:
-    or `:info` (when a terminal is not available).
+    `:auto` (default), `:trace`, `:debug`, `:info`, `:warn`, `:error`,
+    or `:fatal`. The `:auto` level will use `:warn` when a terminal is
+    available or `:info` when a terminal is not available.
```
This may be wrong now, I think we use `:info` by default and `:warn` is only used now when switching to syslog-ing, or when the user explicitly does `--force-logger`.

lamont-granquist replied:
(honestly I'm not sure how to document this right now for Chef-16, that's the best I can come up, I would need to burn a lot of time testing it, and I wouldn't know how to English that up for user consumption since it is a pile of edge conditions nearly nobody should care about any more)

tas50 replied:
@lamont-granquist Wanna toss a suggestion to change that?

tas50 replied:
From Slack "Just ship it"

lamont-granquist commented on content/config_rb_client.md:
```diff
@@ -444,13 +446,16 @@ This configuration file has the following settings:
+    execution of Chef Infra Client's built-in resources. Setting this
+    to true will skip many large and time consuming plugins such as
+    `cloud` or `packages`. Setting this to true may break cookbooks
+    that assume all Ohai data will be present.
```
Runs only the Ohai plugins that determine the O/S platform, hostname, and init system.  This can be used to speed up testing or `chef-apply` scripts, but will break many cookbooks which expect more extended Ohai attributes.

tas50 replied:
That's all there. People don't need to know about chef testing here.

lamont-granquist commented on content/config_rb_client.md:
```diff
@@ -464,11 +469,11 @@ This configuration file has the following settings:
+    authenticating to a Chef Infra Server. By default, Chef Infra Client
+    will use the system's FQDN as the node name. In general, Chef recommends
+    that you leave this setting blank and let the client assign the FQDN of
     the node as the `node_name` during each Chef Infra Client run.
```
Feels like there should be something here about the "unique identity" of the node.

tas50 replied:
Fixed everywhere

tas50 committed:
Clarify node name

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Merge branch 'master' into client.rb
